runtime/legion/legion_tasks.cc

```diff
@@ -3148,6 +3174,51 @@ namespace Legion {
       perform_postmap = output.postmap_task;
     }
 
+    //--------------------------------------------------------------------------
+    void SingleTask::replay_map_task_output(PhysicalTraceInfo &trace_info)
+    //--------------------------------------------------------------------------
+    {
+      std::vector<Processor> procs;
+      trace_info.tpl->get_mapper_output(trace_info, selected_variant,
+          task_priority, perform_postmap, procs, physical_instances);
+
+      if (Runtime::separate_runtime_instances)
+      {
+        target_processors.resize(1);
+        target_processors[0] = this->target_proc;
+      }
+      else // the common case
+        target_processors = procs;
+
+      virtual_mapped.resize(regions.size(), false);
+      std::map<PhysicalManager*,std::pair<unsigned,bool> > *acquired =
+        get_acquired_instances_ref();
+      std::vector<PhysicalManager*> unacquired;
+      for (unsigned idx = 0; idx < regions.size(); idx++)
+      {
+        InstanceSet &instances = physical_instances[idx];
+        if (IS_NO_ACCESS(regions[idx]))
+          continue;
+        if (instances.is_virtual_mapping())
+          virtual_mapped[idx] = true;
+        else
+        {
+          size_t num_instances = instances.size();
+          for (unsigned iidx = 0; iidx < num_instances; ++iidx)
+          {
+            PhysicalManager *manager = instances[iidx].get_manager();
+            if (acquired->find(manager) == acquired->end())
+              unacquired.push_back(manager);
+          }
+        }
+        if (Runtime::legion_spy_enabled)
+          runtime->forest->log_mapping_decision(unique_op_id, idx,
+                                                regions[idx],
+                                                instances);
+      }
+      runtime->forest->perform_missing_acquires(this, *acquired, unacquired);
```

Comment:
Why is this necessary? You should really need this because I'm assuming your trace is already holding references to the physical instances to prevent them from being garbage collected and we shouldn't even be running through the code that checks the mapper output when we are replaying a trace.

Reply:
Pinging you on this again since I don't think you've addressed it yet